pkg/template/api/validation/validation.go

```diff
@@ -51,3 +55,116 @@ func validateTemplateBody(template *api.Template) (allErrs field.ErrorList) {
 	allErrs = append(allErrs, unversionedvalidation.ValidateLabels(template.ObjectLabels, field.NewPath("labels"))...)
 	return
 }
+
+// ValidateTemplate tests if required fields in the Template are set.
+func ValidateTemplateInstance(oc *client.Client, ctx kapi.Context, ti *api.TemplateInstance) (allErrs field.ErrorList) {
```

Comment:
yeah i think your concern is valid, this validation logic should just make sure the username is a valid value (fits whatever regex we enforce on usernames).

After that, if it turns out the user doesn't have the requisite permissions (or doesn't exist), we should find that out when we actually go to instantiate the template as the user, and when it fails, mark the provision request as failed.